src/containers/UserRegistration/home.tsx

```diff
@@ -0,0 +1,280 @@
+      forceConsentPrompt: true, // [Android] if you want to show the authorization prompt at each login.
+      accountName: '', // [Android] specifies an account name on the device that should be used
+      iosClientId:
+        '930089845590-h9l64aorv23u7egacsjt6svi2t4mj3v1.apps.googleusercontent.com', // [iOS] optional, if you want to specify the client ID of type iOS (otherwise, it is taken from GoogleService-Info.plist)
+    });
+    try {
+      await GoogleSignin.hasPlayServices();
+      const userInfo = await GoogleSignin.signIn();
+      const user = {
+        name: userInfo.user.name,
+        email: userInfo.user.email,
+        photo: userInfo.user.photo,
+        id: userInfo.user.id,
+        loginType: 'GOOGLE',
+      };
+      this.setState({user});
+      console.log(user);
+    } catch (error) {
+      if (error.code === statusCodes.SIGN_IN_CANCELLED) {
+        // user cancelled the login flow
+      } else if (error.code === statusCodes.IN_PROGRESS) {
+        // operation (f.e. sign in) is in progress already
+      } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
+        // play services not available or outdated
+      } else {
+        // some other error happened
+      }
+    }
+  };
+
+  loginUsingFacebook = async () => {
+    try {
+      const result = await LoginManager.logInWithPermissions([
+        'public_profile',
+        'email',
+      ]);
+      if (result) {
+        if (result.isCancelled) {
+          console.log('Login cancelled');
+        } else {
+          if (result && result.grantedPermissions) {
+            console.log(
+              'Login success with permissions: ' +
+                result.grantedPermissions.toString(),
+            );
+            const data = await AccessToken.getCurrentAccessToken();
+            if (data) {
+              console.log(data.accessToken.toString());
+              this.initUser(data.accessToken.toString());
+            }
+          }
+        }
+      }
+    } catch (error) {
+      console.log('Login fail with error: ' + error);
+    }
+  };
+
+  initUser(token: string) {
+    fetch(
+      'https://graph.facebook.com/v2.5/me?fields=email,name,picture.type(large)&access_token=' +
+        token,
+    )
+      .then(response => response.json())
+      .then(userInfo => {
+        const user = {
+          name: userInfo.name,
+          email: userInfo.email,
+          photo: userInfo.picture.data.url,
+          id: userInfo.id,
+          loginType: 'FACEBOOK',
+        };
+        this.setState({user});
+        console.log(user);
+      })
+      .catch(() => {
+        console.log('ERROR GETTING DATA FROM FACEBOOK');
+      });
+  }
+
+  render() {
+    return (
+      <SafeAreaView style={Styles.container}>
+        <KeyboardShift>
```

Comment:
Why keyboardshift, and why not KeyBoard avoiding scroll view?

Reply:
Keyboardshift is a component I created that can animate and push the view up even in iOS in a way we need it to be.